microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/integration/FailSafeCDIModelFilePropsHealthTest.java

```diff
@@ -0,0 +1,110 @@
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup({ MicroProfileFTSetupTask.class, FailSafeCDIModelFilePropsHealthTest.SetupTask.class })
+public class FailSafeCDIModelFilePropsHealthTest extends FailSafeCDIHealthBaseTest {
+
+    private byte[] liveBytes;
+    private byte[] readyBytes;
+    private byte[] inMaintananceBytes;
+    private byte[] readyInMaintananceBytes;
+
+    Path liveFile = Paths.get(
+            FailSafeCDIModelFilePropsHealthTest.class.getResource("file-props/" + FailSafeDummyService.LIVE_CONFIG_PROPERTY)
+                    .getPath());
+
+    Path readyFile = Paths.get(
+            FailSafeCDIModelFilePropsHealthTest.class.getResource("file-props/" + FailSafeDummyService.READY_CONFIG_PROPERTY)
+                    .getFile());
+
+    Path inMainenanceFile = Paths.get(
+            FailSafeCDIModelFilePropsHealthTest.class
+                    .getResource("file-props/" + FailSafeDummyService.IN_MAINTENANCE_CONFIG_PROPERTY).getPath());
+
+    Path readyInMainenanceFile = Paths.get(
+            FailSafeCDIModelFilePropsHealthTest.class
+                    .getResource("file-props/" + FailSafeDummyService.READY_IN_MAINTENANCE_CONFIG_PROPERTY).getPath());
+
+    @Deployment(testable = false)
+    public static WebArchive createDeployment() {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, FailSafeCDIModelPropsHealthTest.class + ".war")
+                .addClasses(FailSafeDummyService.class, CDIBasedLivenessHealthCheck.class, CDIBasedReadinessHealthCheck.class)
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+        return war;
+    }
+
+    @Before
+    public void backup() throws IOException {
+        liveBytes = Files.readAllBytes(liveFile);
+        readyBytes = Files.readAllBytes(readyFile);
+        inMaintananceBytes = Files.readAllBytes(inMainenanceFile);
+        readyInMaintananceBytes = Files.readAllBytes(inMainenanceFile);
+    }
+
+    @After
+    public void restore() throws IOException {
+        Files.write(liveFile, liveBytes);
+        Files.write(readyFile, readyBytes);
+        Files.write(inMainenanceFile, inMaintananceBytes);
+        Files.write(readyInMainenanceFile, readyInMaintananceBytes);
+    }
+
+    @Override
+    protected void setConfigProperties(boolean live, boolean ready, boolean inMaintanance, boolean readyInMainenance)
+            throws Exception {
+        Files.write(liveFile, Boolean.toString(live).getBytes(StandardCharsets.UTF_8));
+        Files.write(readyFile, Boolean.toString(ready).getBytes(StandardCharsets.UTF_8));
+        Files.write(inMainenanceFile, Boolean.toString(inMaintanance).getBytes(StandardCharsets.UTF_8));
+        Files.write(readyInMainenanceFile, Boolean.toString(readyInMainenance).getBytes(StandardCharsets.UTF_8));
+        new Administration(ManagementClientProvider.onlineStandalone(managementClient)).reload();
+
```

Comment:
Additional empty line here?
Anyway, here again I had to go to the superclass implementation to see that `setConfigProperties()` is called before each test and so `new Administration(ManagementClientProvider.onlineStandalone(managementClient)).reload();` is meaningful.
This is IMO another signal for a refactoring in order to get the logical flow implemented differently in the classes/hierarchy.

Reply:
well that is because different way of configuring MP Config requires different way of setting the properties. Some ways needs reload to apply the changes some does not.

Reply:
Yes, sure. I get what you mean, a different scenario requires for different actions to be taken. 
My comment here was related to the same argument we mentioned earlier, i.e. inheritance and so on.